servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/GrpcSemanticAttributesExtractor.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2025 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry.http;
+
+import io.servicetalk.transport.api.ConnectionInfo;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import javax.annotation.Nullable;
+
+/**
+ * Extractor for gRPC semantic convention attributes using stable OpenTelemetry APIs.
+ * <p>
+ * This class manually implements RPC semantic conventions as defined by OpenTelemetry,
+ * avoiding dependencies on alpha/incubator APIs.
+ */
+final class GrpcSemanticAttributesExtractor implements AttributesExtractor<RequestInfo, GrpcTelemetryStatus> {
+
+    // RPC semantic convention attribute keys
+    private static final AttributeKey<String> RPC_SYSTEM = AttributeKey.stringKey("rpc.system");
+    private static final AttributeKey<String> RPC_SERVICE = AttributeKey.stringKey("rpc.service");
+    private static final AttributeKey<String> RPC_METHOD = AttributeKey.stringKey("rpc.method");
+    private static final AttributeKey<Long> RPC_GRPC_STATUS_CODE = AttributeKey.longKey("rpc.grpc.status_code");
```

Comment:
Is there any difference in published spans based on an attribute type or all of them look like strings anyway? Even though the spec defines this attribute of type `int`, if it doesn't affect anything else we can keep it as `string` to avoid overhead of conversion and boxing